p2p/protocols/protocol.go

```diff
@@ -290,9 +226,15 @@ func (p *Peer) run(handler func(ctx context.Context, msg interface{}) error) fun
 				defer p.wg.Done()
 				err := p.handleMsg(msg, handler)
 				if err != nil {
-					select {
-					case errc <- err:
-					default:
+					var e *breakError
+					if errors.As(err, &e) {
+						select {
+						case errc <- err:
+						default:
+						}
+					} else {
+						// todo: what to do with errors that are not breaking the loop?
```

Comment:
I think that logging such errors is enough.

Reply:
Debug, or maybe even Info? I would probably not log it as error, as it is not breaking the loop? Or maybe log additional context like `non fatal: err`?

Reply:
I think that Warn would be appropriate.